kapitel/kapitel1.tex

```diff
@@ -0,0 +1,54 @@
+Als Versionsverwaltung für dieses Projekt wird Git, mit GitFlow Workflow, genutzt, speziell der Service GitLab. Mit GitLab wird außerdem eine \ac{CICD} Pipeline realisiert, also eine Möglichkeit automatisch nach jeder Änderung das Projekt zu bauen, zu testen und zu veröffentlichen. Veröffentlicht wird der \ac{OTR} in einem Kubernetes-Cluster, also wird das Projekt vorher als Dockerimage gebaut um es dann in einem Container laufen zu lassen. Für jeden Feature-, Develop- und Masterbranch im Git wird jeweils eine eigene Instanz im Kubernetes Cluster erstellt, so dass man jeden Stand auf dem Testsystem testen kann.
+\\\\
+Bei einem \ac{OTR} hat ein Makler die Möglichkeit Angaben zu einem Versicherungstarif zu treffen und sich anhand dessen Beiträge ausrechnenzulassen. Außerdem sind hilfreiche Funktionalitäten gegeben, wie z.B. das Ausdrucken von Anträgen und Angeboten, das Speichern von Angeboten, um diese später wieder laden und weiter bearbeiten zu können und das Versenden von tarifrelevanten Dokumenten per E-Mail. Diese Funktionalitäten werden in der Fußzeile in Form von Knöpfen angeboten, um dessen Implementierung dieses Projekt handelt. Die Fußzeile beinhaltet außerdem Angaben zum Copyright und einen Verweis zum Impressum.
+\\\\
+
+%Angebotsspeicherung
+Beim Speichern eines Angebotes sollen alle im \ac{OTR} eingegebenen Daten, so wie, wenn anhand der Angaben möglich, berechnete Beiträge im sogennanten Maklerportal für den angemeldeten Makler gespeichert werden. Das Maklerportal ist eine Webseite an der sich Makler anmelden können wo sie Zugriff auf Tarifrechner haben, Übersichten über ihr Kundenbestand und noch vieles mehr. Zu dem können dort also auch gespeicherte Angebote dort eingesehen, bearbeitet und mit einem \ac{OTR} aufgerufen werden. Beim Speichern wird über mehrere anliegende Systeme ein Angebot in einer Datenbank gespeichert. Wichtig ist dabei die Session, welche von einem der Systeme generiert wird, realisiert als \ac{GUID}. Mithilfe dieser wird ein Angebot im XML-Format in einer Datenbank gespeichert und ist dadurch eindeutig identifizierbar.\\
+Außerdem ist es vorgesehen, dass Makler Angebote kopieren können. Dies beinhaltet die gleiche Logik wie die Angebotsspeicherung, bis auf dass eine neue \ac{GUID} erzeugt wird, wodurch dementsprechend ein neuer Eintrag in der Datenbank und im Maklerportal hinterlegt wird. Außerdem muss mit der \ac{GUID}, bzw. Session der Kopie weitergearbeitet werden.\\
```

Comment:
zu Detailiert, ich würde dies auch ehr in der Implementierung sehen, hier nur sowas wie Kopie von gespeicherten Angeboten soll möglich sein, ist ähnlich des Speichern von Angebote und jeder braucht eine eindeutige Nummer zum Identifizieren.